app/main.c

```diff
@@ -267,7 +269,13 @@ static void global_init(void)
     num_enabled_cpus = 0;
     for (int i = 0; i < num_available_cpus; i++) {
         if (cpu_state[i] == CPU_STATE_ENABLED) {
-            chunk_index[i] = num_enabled_cpus;
+            if (enable_numa) {
+                uint32_t proximity_domain_idx = smp_get_proximity_domain_idx(i);
+                chunk_index_in_proximity_domain[i] = smp_alloc_cpu_in_proximity_domain(proximity_domain_idx);
```

Comment:
Couldn't you have just used `chunk_index` for this case too?

Reply:
In the end, yes, it seems so. At first, I wasn't sure it would be possible not to use both `chunk_index` and `chunk_index_in_proximity_domain` in the same code path.

Reply:
Got rid of `chunk_index_in_proximity_domain`.